odoo/addons/base/tests/test_res_users.py

```diff
@@ -190,6 +190,21 @@ def test_deactivate_portal_users_archive_and_remove(self):
         self.assertTrue(portal_partner_2.exists(), 'Should have kept the partner')
         self.assertEqual(asked_deletion_2.state, 'fail', 'Should have marked the deletion as failed')
 
+    @mute_logger('odoo.sql_db')
+    def test_delete_public_user(self):
+        """Test that the public user cannot be deleted."""
+        User = self.env['res.users']
+        public_user = self.env.ref('base.public_user')
+        public_partner = public_user.partner_id
+
+        # Attempt to delete the public user
+        with self.assertRaises(UserError, msg='Public user should not be deletable'):
+            public_user.unlink()
+
+        # Ensure the public user still exists and is active
+        self.assertTrue(public_user.exists() and public_user.active, 'Public user should still exist and be active')
+        self.assertTrue(public_partner.exists() and public_partner.active, 'Public partner should still exist and be active')
```

Comment:
This should fail because Public User and Public Partner are active=False by default.